Deprecate Contentstore for Video Component Preview LMS/CMS

muhammad-ammar opened the pull request.

muhammad-ammar commented:
jenkins run all

muhammad-ammar commented:
@Qubad786 Can you give it a quick look?

Qubad786 commented on common/lib/xmodule/xmodule/video_module/video_handlers.py:
```diff
@@ -297,69 +306,46 @@ def transcript(self, request, dispatch):
-                # Check for transcripts in edx-val as a last resort if corresponding feature is enabled.
-                if feature_enabled:
-                    # Make sure the language is set.
-                    if not lang:
```
@muhammad-ammar if lang is not set we will assume a default language, this check should be preserved

Qubad786 replied:
And I think this is contentstore interface level change :-/

Qubad786 commented on lms/djangoapps/mobile_api/video_outlines/views.py:
```diff
@@ -120,28 +121,10 @@ def get(self, request, course, *args, **kwargs):
-            mimetype = Transcript.mime_types[Transcript.SRT]
-        except KeyError:
-            raise Http404(u"Transcript not found for {}, lang: {}".format(block_id, lang))
+            content, filename, mimetype = get_transcript(course.id, block_id, lang)
```
Nice!

muhammad-ammar commented:
@Qubad786 Addressed your feedback

muhammad-ammar commented:
@Qubad786 Addressed your feedback. Also separated the bumper transcripts handling in `translations` dispatch in `video_handlers.py`. Please have a look.

Qubad786 commented on common/lib/xmodule/xmodule/video_module/transcripts_utils.py:
```diff
@@ -973,29 +974,33 @@ def get_transcript_from_contentstore(video, language, output_format, youtube_id=
+        edx_video_id = clean_video_id(video.edx_video_id)
+        if not edx_video_id:
+            raise NotFoundError
+        return get_transcript_from_val(video.edx_video_id, lang, output_format)
```
pass cleaned `edx_video_id` into the VAL.

Qubad786 commented on common/lib/xmodule/xmodule/video_module/transcripts_utils.py:
```diff
@@ -973,29 +974,33 @@ def get_transcript_from_contentstore(video, language, output_format, youtube_id=
     """
-    usage_key = BlockUsageLocator(course_id, block_type='video', block_id=block_id)
-    video_descriptor = modulestore().get_item(usage_key)
+    transcripts_info = video.get_transcripts_info()
```
Also, `include_val_transcripts=True`.

Qubad786 commented on common/lib/xmodule/xmodule/video_module/video_handlers.py:
```diff
@@ -288,79 +314,50 @@ def transcript(self, request, dispatch):
+                    self,
+                    lang=self.transcript_language,
+                    output_format=Transcript.SJSON,
+                    youtube_id=request.GET.get('videoId', None),
```
`request.GET.get('videoId')` would do.

Qubad786 commented on common/lib/xmodule/xmodule/video_module/video_handlers.py:
```diff
@@ -288,79 +314,50 @@ def transcript(self, request, dispatch):
+                    mimetype
+                )
+            except NotFoundError as ex:
+                log.info(six.text_type(ex))
```
Or we can use `log.exception('this is %s exc',  self.location)`

Qubad786 commented on common/lib/xmodule/xmodule/video_module/video_handlers.py:
```diff
@@ -246,6 +248,29 @@ def publish_completion(self, data, dispatch):  # pylint: disable=unused-argument
+        response = Response(
+            content,
+            headerlist=[
+                ('Content-Disposition', 'attachment; filename="{}"'.format(filename.encode('utf-8'))),
```
We don't need transcript as an attachment in case of `/translations/{lang}` dispatch.

Qubad786 commented on common/lib/xmodule/xmodule/video_module/video_handlers.py:
```diff
@@ -288,79 +314,50 @@ def transcript(self, request, dispatch):
-                    transcripts, transcript_format=self.transcript_download_format, lang=lang
-                )
-            except (KeyError, UnicodeDecodeError):
+                content, filename, mimetype = get_transcript(self, lang)
```
we can construct response here, to avoid multiple return statements:
```bash
response = self.make_transcript_http_response(
                content,
                filename,
                self.transcript_language,
                mimetype
            )
```

Qubad786 commented on common/lib/xmodule/xmodule/video_module/video_handlers.py:
```diff
@@ -288,79 +314,50 @@ def transcript(self, request, dispatch):
+                        transcripts
+                    )
+
+                    return self.make_transcript_http_response(
```
Can we simplify this to have less `return` statements?

Qubad786 commented on common/lib/xmodule/xmodule/video_module/video_handlers.py:
```diff
@@ -273,6 +305,7 @@ def transcript(self, request, dispatch):
         # Currently, we don't handle video pre-load/bumper transcripts in edx-val.
         feature_enabled = is_val_transcript_feature_enabled_for_course(self.course_id) and not is_bumper
         transcripts = self.get_transcripts_info(is_bumper, include_val_transcripts=feature_enabled)
```
We should not include VAL transcripts in case of bumper.
```bash
 self.get_transcripts_info(is_bumper, include_val_transcripts=not is_bumper)
```

Qubad786 commented on common/lib/xmodule/xmodule/video_module/video_handlers.py:
```diff
@@ -288,79 +321,45 @@ def transcript(self, request, dispatch):
-                    transcripts, transcript_format=self.transcript_download_format, lang=lang
-                )
-            except (KeyError, UnicodeDecodeError):
+                content, filename, mimetype = get_transcript(self, lang)
```
What about bumper case?

Qubad786 commented on common/lib/xmodule/xmodule/video_module/video_handlers.py:
```diff
@@ -288,79 +321,45 @@ def transcript(self, request, dispatch):
+                )
+            except NotFoundError:
+                log.exception('[Translation Dispatch] %s', self.location)
+                return self.get_static_transcript(request, transcripts)
```
we can avoid this return statement by:
```bash
response = self.get_static_transcript(request, transcripts)
```

muhammad-ammar commented:
@Qubad786 made more changes. have a look

Qubad786 commented on lms/djangoapps/courseware/tests/test_video_handlers.py:
```diff
@@ -800,33 +814,6 @@ def test_translation_fallback_transcript_feature_disabled(self):
-    @ddt.data(True, False)
-    @patch('xmodule.video_module.transcripts_utils.edxval_api.get_video_transcript_data')
-    @patch('openedx.core.djangoapps.video_config.models.VideoTranscriptEnabledFlag.feature_enabled')
-    def test_translations_bumper_transcript(self, feature_enabled,
```
why remove this?

Qubad786 commented on common/lib/xmodule/xmodule/video_module/transcripts_utils.py:
```diff
@@ -973,29 +974,33 @@ def get_transcript_from_contentstore(video, language, output_format, youtube_id=
     """
-    usage_key = BlockUsageLocator(course_id, block_type='video', block_id=block_id)
-    video_descriptor = modulestore().get_item(usage_key)
+    transcripts_info = video.get_transcripts_info(include_val_transcripts=True)
```
haven't we removed `include_val_transcripts=True`?

muhammad-ammar replied:
this is removed

Qubad786 commented on common/lib/xmodule/xmodule/video_module/video_handlers.py:
```diff
@@ -288,85 +319,52 @@ def transcript(self, request, dispatch):
+            )
         elif dispatch.startswith('available_translations'):
-
+            feature_enabled = is_val_transcript_feature_enabled_for_course(self.course_id) and not is_bumper
```
This will go away right?

muhammad-ammar replied:
yes, in this https://github.com/edx/edx-platform/pull/17672
